langchain/src/vectorstores/closevector_node.ts

```diff
@@ -0,0 +1,286 @@
+import {
+  CloseVectorHNSWNode,
+  HierarchicalNSWT,
+  CloseVectorHNSWLibArgs,
+  CloseVectorDocument,
+  CloseVectorCredentials,
+} from "closevector-node";
+
+import { Embeddings } from "../embeddings/base.js";
+import { Document } from "../document.js";
+import { SaveableVectorStore } from "./base.js";
+
+/**
+ * package closevector-node is largely based on hnswlib.ts in the current folder with the following exceptions:
+ * 1. It uses a modified version of hnswlib-node to ensure the generated index can be loaded by closevector_web.ts.
+ * 2. It adds features to upload and download the index to/from the CDN provided by CloseVector.
+ *
+ * For more information, check out https://closevector-docs.getmegaportal.com/
+ */
+
+/**
+ * Arguments for creating a CloseVectorNode instance, extending CloseVectorHNSWLibArgs.
+ */
+export interface CloseVectorNodeArgs
+  extends CloseVectorHNSWLibArgs<HierarchicalNSWT> {
+  instance?: CloseVectorHNSWNode;
+}
+
+/**
+ * Class that implements a vector store using Hierarchical Navigable Small
+ * World (HNSW) graphs. It extends the SaveableVectorStore class and
+ * provides methods for adding documents and vectors, performing
+ * similarity searches, and saving and loading the vector store.
+ */
+export class CloseVectorNode extends SaveableVectorStore {
+  declare FilterType: (doc: Document) => boolean;
+
+  _instance?: CloseVectorHNSWNode;
```

Comment:
Why `this._instance` and `this.instance`?

Reply:
I was thinking about making the setter private to prevent the instance from being changed from the outside.